src/Graph.js

```diff
@@ -158,6 +160,43 @@ prototype.propagate = function(pulse, node, stamp) {
   }
 };
 
+// Process a new branch of the dataflow graph prior to connection:
+// (1) Insert new Collector nodes as needed. 
+// (2) Track + return mutation/routing status of the branch.
+prototype.preprocess = function(branch) {
+  var graph = this,
+      mutates = 0,
+      node, router, collector, collects;
```

Comment:
Right, in a general call to this method, we will insert a collector. However, when called from `DataSource`, `_inputNode` is unshifted onto `pipeline`, which is passed to `preprocess`. The `_inputNode` is marked as a collector and, being the first node in the `branch`, should yield the same behaviour I think.